### PR TITLE
Fix redundant OnPermissionChanged notification

### LINKED ISSUE #3004: SDL sends OnPermissionsChange twice
# Bug Report
SDL sends OnPermissionsChange twice

##### Reproduction Steps
1. Start SDL, HMI, connect Mobile, register new App => PTU is started
2. Wait until PTU is finished successfully

##### Expected Behavior
SDL sends OnPermissionsChange to App only once

##### Observed Behavior
SDL sends OnPermissionsChange to App twice

##### OS & Version Information
* OS/Version: Ubuntu 16.04 amd64
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (e7b5c75)
* ATF: https://github.com/smartdevicelink/sdl_atf/tree/develop
* Testing Against: ATF (script, report and logs are in attachment)

##### Test Case, Sample Code, and / or Example App
- [RPCMessageProtection/024_1_Params_in_OnPermissionChange_PTU_same_FG](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Security/RPCMessageProtection/024_1_Params_in_OnPermissionChange_PTU_same_FG.lua)
- [build_options/077_ATF_PTU_Changes_Applied_OnPermissionChange_Notification_HMI_HTTP](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Policies/build_options/077_ATF_PTU_Changes_Applied_OnPermissionChange_Notification_HMI_HTTP.lua)
- [build_options/078_ATF_PTU_HMI_Level_Affected_Apps_FULL_LIMITED_HTTP](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Policies/build_options/078_ATF_PTU_HMI_Level_Affected_Apps_FULL_LIMITED_HTTP.lua)

- [2019-09-01_01-17-03.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3563118/2019-09-01_01-17-03.tar.gz)
- [2019-09-01_01-22-08.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3563119/2019-09-01_01-22-08.tar.gz)
- [2019-09-01_01-30-41.tar.gz](https://github.com/smartdevicelink/sdl_core/files/3563122/2019-09-01_01-30-41.tar.gz)

### PR DESCRIPTION
**Pull request by mkorniichuk**
Fixes #3004

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
OnPermissionChange used to be sent from CheckAppPolicy. But now it is sent from PolicyManager. And the duplication of this notification was left in CheckAppPolicy for proprietary flow.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by alexkutsan** [comment]
@mkorniichuk  Provide more detailed description

File: src/components/policy/policy_external/src/policy_manager_impl.cc [hunk 1]
```diff
@@ -574,6 +570,10 @@ CheckAppPolicyResults PolicyManagerImpl::CheckPermissionsChanges(
     const std::shared_ptr<policy_table::Table> snapshot) {
   LOG4CXX_AUTO_TRACE(logger_);
 
+  // Replace predefined policies with its actual setting, e.g. "123":"default"
+  // to actual values of default section
+  UnwrapAppPolicies(pt_update->policy_table.app_policies_section.apps);
```
Comment (JackLivio) [inline src/components/policy/policy_external/src/policy_manager_impl.cc]: UnwrapAppPolicies is moved in the external policies, but below in regular the method is removed?

Comment (mkorniichuk) [reply src/components/policy/policy_external/src/policy_manager_impl.cc]: @JackLivio `UnwrapAppPolicies` is already in `CheckPermissionsChanges` for policy_regular. So the removed one is redundant.

**Comment by JackLivio** [comment]
I am seeing a smoke test failure for this pr

```
./test_scripts/Smoke/Policies/001_PTU_all_flows.lua
```

@mkorniichuk Could you please give this test a try?

**Comment by mkorniichuk** [comment]
@JackLivio sure, I'll take a look at it tomorrow and let you know as soon as possible.

**Comment by JackLivio** [comment]
Also I am experiencing a unit test failure in the policy_test

**Comment by mkorniichuk** [comment]
@JackLivio we've fixed smoke and unit tests. Please take a look.
Also, if everything is ok, please, let us know before merge, we will squash the fixup commit.

**Comment by JackLivio** [comment]
Thank you for the update. Has this pr been tested against External policies? I ask because there were a considerable amount of changes in the last PR to "fix" regular policies and there were no related changes submitted for external policies.

**Comment by mkorniichuk** [comment]
@JackLivio we've tested unit tests and Smoke ATF tests for proprietary and external proprietary flow.
And you're right, there are no much changes for external flow because the issue is related to proprietary flow.

**Comment by JackLivio** [comment]
```
./test_scripts/Policies/App_Permissions/001_ATF_HP_Order_Of_Request_Response_Notification_On_Registering_App.lua
./test_scripts/Policies/App_Permissions/002_ATF_HP_OnPermissionsChange_With_Assigned_Policy_After_App_Registration.lua
./test_scripts/Policies/App_Permissions/022_ATF_No_Permission_Notification_To_HMI_In_First_App_Registration.lua
```

These tests fail with OnPermissionsChange related Error Logs.

ie
```
OnPermissionsChange is not received 4 in message order. Real: received number: 3 
```

Using External Proprietary build config.

**Comment by dboltovskyi** [comment]
@JackLivio Could you double check please on your side since we were not able to reproduce the issues you mentioned on our environment with the latest sources:
https://github.com/smartdevicelink/sdl_core/tree/fix/SDL_sends_OnPermissionsChange_twice (861b28e)
https://github.com/smartdevicelink/sdl_atf_test_scripts/tree/develop (ebc9063)

However there is known instability in `App_Permissions/001` script and the message you mentioned is related to it:
> OnPermissionsChange is not received 4 in message order. Real: received number: 3 

It's about the order of messages during app registration and not the number of `OnPermissionsChange` notifications.
This issue is fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2234

**Pull request merged**